src/cascadia/TerminalSettings/KeyChord.cpp

```diff
@@ -39,4 +115,171 @@ namespace winrt::Microsoft::Terminal::Settings::implementation
     {
         _vkey = value;
     }
+
+    // Method Description:
+    // - Deserializes the given string into a new KeyChord instance. If this
+    //   fails to translate the string into a keychord, it will throw a
+    //   hresult_invalid_argument exception.
+    // - The string should fit the format "[ctrl+][alt+][shift+]<keyName>",
+    //   where each modifier is optional, and keyName is either one of the
+    //   names listed in the vkeyNamePairs vector above, or is one of 0-9a-zA-Z.
+    // Arguments:
+    // - hstr: the string to parse into a keychord.
+    // Return Value:
+    // - a newly constructed KeyChord
+    winrt::Microsoft::Terminal::Settings::KeyChord KeyChord::FromString(const hstring& hstr)
+    {
+        std::wstring wstr{ hstr };
+
+        // Split the string on '+'
+        std::wstring temp;
+        std::vector<std::wstring> parts;
+        std::wstringstream wss(wstr);
+
+        while(std::getline(wss, temp, L'+'))
+        {
+            parts.push_back(temp);
+        }
+
+        // If we have > 4, something's wrong.
+        if (parts.size() > 4)
+        {
+            throw hresult_invalid_argument();
+        }
+
+        KeyModifiers modifiers = KeyModifiers::None;
+        int32_t vkey = 0;
+
+        // Look for ctrl, shift, alt. Anything else might be a key
+        for (const auto& part : parts)
+        {
+            if (part == CTRL_KEY)
+            {
+                modifiers |= KeyModifiers::Ctrl;
+            }
+            else if (part == ALT_KEY)
+            {
+                modifiers |= KeyModifiers::Alt;
+            }
+            else if (part == SHIFT_KEY)
```

Comment:
Do we have to worry about left-shift vs right-shift? Do we want to?

Reply:
We recently learned that right alt is altgr sometimes?

Reply:
This is true, but I'm not sure UWP can differentiate between the two in the KeyEvent. I'm pretty sure there's only a Ctrl, Shift, and Alt, and not a left/right version :/

If we find out otherwise, then the entire KeyModifiers structure might need to be updated, as well as any consumers.